sklearn/model_selection/_split.py

```diff
@@ -1100,6 +1100,127 @@ def split(self, X, y=None, groups=None):
                     indices[test_start : test_start + test_size],
                 )
 
+class TimeSeriesInitialSplit(_BaseKFold):
+    """Time Series cross-validator with initial period
+
+    Provides train/test indices to split time series data samples
+    that are observed at fixed time intervals, in train/test sets.
+    In each split, test indices must be higher than before, and thus shuffling
+    in cross validator is inappropriate. This is distinct from the
+    TimeSeriesSplit in that users specify the initial training period
+    which is normally larger than the test block size.
+
+    Note that unlike standard cross-validation methods, successive
+    training sets are supersets of those that come before them.
+
+    Read more in the :ref:`User Guide <time_series_initial_split>`.
+
+    .. versionadded:: 1.1
+
+    Parameters
+    ----------
+    initial : int, default=21
+        Number of splits.
+
+    max_train_size : int, default=None
+        Maximum size for a single training set.
+
+    test_size : int, default=7
+        Sets the size of the test set to be added at each iteration
+
+    gap : int, default=0
+        Number of samples to exclude from the end of each train set before
+        the test set.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.model_selection import TimeSeriesInitialSplit
+    >>> X = np.arange(0,50)
+    >>> tscv = TimeSeriesInitialSplit()
+    >>> for train_index, test_index in tscv.split(X):
+    ...     print("TRAIN:", train_index, "TEST:", test_index)
+    ...     X_train, X_test = X[train_index], X[test_index]
+    ...     y_train, y_test = y[train_index], y[test_index]
+    TRAIN: [0] TEST: [1]
+    TRAIN: [0 1] TEST: [2]
+    TRAIN: [0 1 2] TEST: [3]
+    TRAIN: [0 1 2 3] TEST: [4]
+    TRAIN: [0 1 2 3 4] TEST: [5]
```

Comment:
Thanks for the response. I should explain this better.

![image](https://user-images.githubusercontent.com/9298693/180017314-70be5d73-492b-49dd-b130-f09a5e64a176.png)

Assume the blue dots represents dates to train on and the red dots are for testing.

In this picture the initial size of the blue dots are much larger than the size of the testing dots. (6 blue dots vs 1 red). With each iteration train size increases by 1 blue dot. 

My PR allows this kind of flexibility. A user may set a larger initial training period, which is not bound by the size that the training set is increases by on each iteration. It decouples the initial size from the increment size.

Reply:
> I am assuming people could use this function in a forecast setting for daily forecasts.

This assumption is not always valid. I've seen the time series splitter used for hourly data, etc.

In that case with `initial=21`, should the first split have training indices `[0, 1, 2, 3, 4, ..., 20]`, and the test set be `[21, 22, .., 27]` because `test_size=7`?

It looks like you want a walk forward as in https://github.com/scikit-learn/scikit-learn/pull/23780 but with an additional `min_train_size`.

Reply:
That is fair enough. I am not married to `initial=21`. You let me know what is preferred and I will make the change. The functionality is much more important to me than the initial args.

Reply:
If you sign off on this then I can keep working. I remember seeing a contributors document. I am would have to see what the expectation is for writing tests and documentation.

Reply:
I am not signing off on this because the API is not finalized. Let's move the discussion to the issue.